src/app/icd/IcdMonitoringTable.cpp

```diff
@@ -67,7 +88,29 @@ void IcdMonitoringEntry::Clear()
 {
     this->checkInNodeID    = kUndefinedNodeId;
     this->monitoredSubject = kUndefinedNodeId;
-    this->key              = ByteSpan();
+    if (sessionKeyStore != nullptr)
```

Comment:
We should return an error and fail if we don't have the sessionKeyStore.

Reply:
Well that's not possible since this function is an override, we cannot return any errors whatsoever. However the check are done before so we should be good in most cases.